tests/test_execution.py

```diff
@@ -5,22 +5,14 @@
 import substratest as sbt
 
 
-def test_tuples_execution_on_same_node(factory, session):
+def test_tuples_execution_on_same_node(data_network):
     """Execution of a traintuple, a following testtuple and a following traintuple."""
-    spec = factory.create_dataset()
-    dataset = session.add_dataset(spec)
+    factory, network = data_network
+    session = network.sessions[0]
 
-    spec = factory.create_data_sample(test_only=True, datasets=[dataset])
-    test_data_sample = session.add_data_sample(spec)
-
-    spec = factory.create_data_sample(test_only=False, datasets=[dataset])
-    train_data_sample = session.add_data_sample(spec)
-
-    spec = factory.create_objective(
-        dataset=dataset,
-        data_samples=[test_data_sample],
-    )
-    objective = session.add_objective(spec)
+    dataset = session.state.datasets[0]
+    objective = session.state.objectives[0]
+    train_data_samples = session.state.train_data_samples
```

Comment:
In all the tests we could remove this line and use directly `dataset.train_data_sample_keys`. Here we assume that all the train data samples in the session state belong to the same dataset.

Reply:
Well, they do :) But you're right, it'd be much more explicit. We'll just need to make sure the datasets do reference all the traintuple keys, and this is exactly the point of another of your comment. I'll make the change.